hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -670,6 +676,7 @@ protected void complete(Object value) {
         if (context.invocationRegistry.deregister(this) && taskDoneCallback != null) {
             context.asyncExecutor.execute(taskDoneCallback);
         }
+        context.invocationRegistry.retire(this);
```

Comment:
Do you need to add this to `completeExceptionally` as well?

Reply:
Fixed